### DISCUSSION TITLE
Use of ServerAction is triggering "ReferenceError: FormData is not defined" while building the application

### DISCUSSION THREAD

**Discussion by kirankulkarni** [vercel/next.js#57983] [Help] [answered]
### Summary

I have been building and deploying my NextJS 13 app for a while now. I adopted server-actions to submit form. 
Now if I run `npm run build` I receive error `ReferenceError: FormData is not defined` in the `Generating static pages` step. 

I tried using `import { experimental_useFormState as useFormState } from "react-dom";` in the client component to use the server action and maintain a state, but did not help

I am using Node v16.15.0 and NextJS 13.5.6

How do I fix this problem

**Comment by kirankulkarni** [comment]
I have created a sample version of the problem https://github.com/kirankulkarni/test-server-actions/tree/main/src/app 

When I run `next run build` it fails with 

```
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
ReferenceError: FormData is not defined
```

**Comment by icyJoseph** [comment, marked as answer]
If I remember correctly, Node 16 doesn't have a  FormData API.

You need at least Node 18, which introduced the fetch family of API's, included FormData.

**Comment by kirankulkarni** [comment]
Yes! upgrading Node to v18.18.2 fixed this problem
